pytest/tests/loadtest/locust/common/sweat.py

```diff
@@ -0,0 +1,174 @@
+
+    def register_oracle(self, node: NearNodeProxy, oracle_id: str):
+        """
+        Register an account as oracle to give it the power to mint tokens for steps.
+        """
+        node.send_tx_retry(SweatAddOracle(self.account, oracle_id),
+                           "add sweat oracle")
+
+
+class InitSweat(FunctionCall):
+
+    def __init__(self, sweat_account: Account):
+        super().__init__(sweat_account, sweat_account.key.account_id, "new")
+
+    def args(self) -> dict:
+        # Technical details about Sweat contract initialization:
+        #
+        # A postfix is used by the smart contract to decide whether or not to
+        # hash an account id. It's an optimization that makes storage keys for
+        # implicit accounts shorter, while preserving short keys of named
+        # accounts.
+        #
+        # More specifically, any account id that matches .*<postfix> will be
+        # stored in the trie normally, like in a unmodified FT contract. Any
+        # other account ids are hashed.
+        #
+        # As an example, the postfix can be `.u.sweat.testnet`.
+        #
+        # Source code for reference:
+        # https://github.com/sweatco/near-sdk-rs/blob/af6ba3cb75e0bbfc26e346e61aa3a0d1d7f5ac7b/near-contract-standards/src/fungible_token/core_impl.rs#L249-L259
+        #
+        # Here we don't provide a postfix, so everything will be hashed. This is
+        # fine for new contracts we create. And when we reuse a contract, we
+        # won't need to initialise it at all.
+        return {"postfix": None}
+
+
+class SweatAddOracle(FunctionCall):
+    """
+    Oracle accounts are allowed to mint new tokens and can only be added by the
+    account id  of the contract itself.
+    """
+
+    def __init__(self, sweat_account: Account, oracle_id: str):
+        super().__init__(sweat_account, sweat_account.key.account_id,
+                         "add_oracle")
+        self.oracle_id = oracle_id
+
+    def args(self) -> dict:
+        return {"account_id": self.oracle_id}
+
+
+class SweatMint(FunctionCall):
+    """
+    A call to `tge_mint`.
+    Token Generation Event (TGE) was day 0 when SWEAT launched.
+    This is the transaction to get initial balance into accounts.
+    """
+
+    def __init__(self, sweat: Account, user_id: str, amount: int):
+        super().__init__(sweat, sweat.key.account_id, "tge_mint")
+        self.user_id = user_id
+        self.amount = amount
+
+    def args(self) -> dict:
+        return {
+            "account_id": self.user_id,
+            "amount": f"{self.amount}",
+        }
+
+
+class SweatMintBatch(FunctionCall):
+    """
+    A call to `record_batch`.
+    Mints new tokens for walked steps for a batch of users.
+    """
+
+    def __init__(self, sweat_id: str, oracle: Account,
+                 recipient_step_paris: list[list[str, int]]):
+        super().__init__(oracle, sweat_id, "record_batch")
+        self.recipient_step_paris = recipient_step_paris
```

Comment:
Do you need to walk in Paris specifically to mint SWEAT? :D
...or it's just a misspelled "pairs"?

Reply:
Haha, good one, yeah that's a misspelling :)